charts/etcd-cluster-operator/templates/etcdcluster_cr.yaml

```diff
@@ -48,16 +48,24 @@ spec:
                 - "{{ .Values.cluster.nodeSelectorTerm.value }}"
       {{- end }}
       podAntiAffinity:
+        requiredDuringSchedulingIgnoredDuringExecution:
+          - labelSelector:
+              matchExpressions:
+                - key: etcd.improbable.io/cluster-name
+                  operator: In
+                  values:
+                    - {{ .Values.cluster.name }}
+            topologyKey: kubernetes.io/hostname
         preferredDuringSchedulingIgnoredDuringExecution:
-          - weight: 100
-            podAffinityTerm:
-              labelSelector:
-                matchExpressions:
-                  - key: etcd.improbable.io/cluster-name
-                    operator: In
-                    values:
-                      - {{ .Values.cluster.name }}
-              topologyKey: kubernetes.io/hostname
+        - weight: 1
+          podAffinityTerm:
+            labelSelector:
+              matchExpressions:
+              - key: etcd.improbable.io/cluster-name
+                operator: In
+                values:
+                  - storageos-etcd
+            topologyKey: topology.kubernetes.io/zone
```

Comment:
You might want to parameterise this to allow users to set their own "topology key" value